text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,270 @@
+# Meta
+[meta]: #meta
+- Name: Pack Publish Buildpack
+- Start Date: 2020-04-27
+- Author(s): [Joe Kutner](https://github.com/jkutner), [Javier Romero](https://github.com/jromero)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md
+
+# Summary
+[summary]: #summary
+
+This RFC describes the implementation the following commands:
+* `pack publish-buildpack`
+* `pack yank-buildpack`
+
+# Motivation
+[motivation]: #motivation
+
+The [Client-Side Registry RFC (0022)](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md) established that two CLI commands would allow users to add/remove buildpacks in the registry, but the implementation of those commands was vauge. In this document we describe the details of how they will work.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Adding a Buildpack to the Registry
+
+When a buildpack author would like to publish a version of a buildpack on the registry, they will use the `pack publish-buildpack <url>` command. This command will create a Github Issue on the `https://github.com/buildpack/registry` (or similarly named) Github repo requesting the addition the new buildpack version to the index.
+
+## Yanking a Buildpack from the Registry
+
+Sometimes a buildpack author may have pushed up a bad version that they wish to not be available in the index. In order to not break builds, it will not be possible to fully remove an entry from the index. Instead, the entry in the index will be marked as "yanked". This information can then be used when resolving which buildpacks to fetch.
```

Comment:
From "How it Works" it looks like this also creates an issue. Can we add that to here as well?

Reply:
added below this line